service/history/api/updateworkflow/api.go

```diff
@@ -76,15 +77,20 @@ func Invoke(
 		return nil, consts.ErrWorkflowExecutionNotFound
 	}
 
-	upd, duplicate, removeFn := weCtx.GetContext().UpdateRegistry().Add(req.GetRequest().GetRequest())
-	if removeFn != nil {
-		defer removeFn()
+	updateID := req.GetRequest().GetRequest().GetMeta().GetUpdateId()
```

Comment:
Request.Request is because the pattern used is to wrap a workflowservice request with a historyservice request. This is pretty common in our system. There's no NPE concern because there's upstream validation (in Frontend service) and because the Go codegen for protobuf types includes a nil check in the Get\<Thing\>() function definitions.

Reply:
`This is pretty common in our system` is a wrong argument IMO ;)
And from the user's perspective, it looks like a bug having Request.Request.
IMO it should be something like this:
```
req.GetWorkflowServiceRequest()
req.GetHistoryServiceRequest()
```

Reply:
Something smells here. You won't make me believe that having `.GetRequest().GetRequest()` is the correct approach.
Shouldn't WorkflowService be able to return `UpdateId` without knowing the internals of the n-th level dependency?

Reply:
What I meant is that wrapping an api request in a history request is common

Reply:
Well, I am happy with `Id`. I think we should change all `ID` in our go code base to `Id`.